projectq/ops/_gates.py

```diff
@@ -338,3 +338,57 @@ def get_inverse(self):
 
 #: Shortcut (instance of) :class:`projectq.ops.BarrierGate`
 Barrier = BarrierGate()
+
+
+class FlipBits(SelfInverseGate):
+    """ Gate for transforming qubits in state |0> to any computational basis state """
+    def __init__(self, bits_to_flip):
+        """
+        Initialize FlipBits gate.
+
+        Example:
+            .. code-block:: python
+
+                qureg = eng.allocate_qureg(2)
+                FlipBits([0, 1]) | qureg
+
+        Note:
+            The amplitude of state k is final_state[k]. When the state k is
+            written in binary notation, then qureg[0] denotes the qubit
+            whose state corresponds to the least significant bit of k.
+
+        Args:
+            bits_to_flip(list[int]|list[bool]|str|int): int or array of 0/1, True/False, or
+                                    string of 0/1 identifying
+                                    the qubits to flip. In case of int, the bits to flip are
+                                    determined from the binary digits, with the least significant
+                                    bit corresponding to qureg[0]. If bits_to_flip is negative,
+                                    exactly all bits which would not be flipped for the input
+                                    -bits_to_flip are flipped.
+        """
+        if isinstance(bits_to_flip, int):
+            self.bits_to_flip = bits_to_flip
+        else:
+            self.bits_to_flip = 0
+            for c in reversed(list(bits_to_flip)):
+                bit = 0b1 if c == '1' or c == 1 or c == True else 0b0
+                self.bits_to_flip = (self.bits_to_flip << 1) | bit
+
+    def __str__(self):
+        return "FlipBits("+str(self.bits_to_flip)+")"
+
+    def __or__(self, qubits):
+        if isinstance(qubits, tuple) and len(qubits) > 1:
+            raise ValueError(self.__str__()+' can only be applied to qubits, quregs, arrays of qubits, and tuples with one individual qubit')
+        for qureg in self.make_tuple_of_qureg(qubits):
+            for i, qubit in enumerate(qureg):
+                if ((self.bits_to_flip if self.bits_to_flip >= 0 else self.bits_to_flip-1) >> i) & 1:
```

Comment:
This should just be `if (self.bits_to_flip >> i) & 1:`

Reply:
I don't think so. The bit patterns of n and -n are not complementary, those of n and -n-1 are.

Reply:
Why do you want them to be complementary? I'd expect `FlipBits` to flip the 1-bits in the two's complement representation of a negative integer, i.e., -1 flips all bits.

Reply:
I guess both variants "make sense", but I agree that having a gate that flips all bits is a very good argument in favor of the behavior your suggest. I made the corresponding change in  30103c7